### ISSUE TITLE
How can Transaction.statusHistory be used to replace Transaction.processorResponse

### ISSUE THREAD

**Issue by lukehedger** [braintree/graphql-api#13]
The schema includes a [deprecation notice on the `Transaction.processorResponse` field](https://github.com/braintree/graphql-api/blob/master/schema.graphql#L5449-L5450). It states "Use relevant events in `statusEvents` instead." but it is unclear how to obtain the processor decline event data, in particular the decline code. (I think `statusEvents` should actually also be `statusHistory`.)

Given the following Sandbox operation, how can I replace `Transaction.processorResponse` with `Transaction.statusHistory` to return the same data in the response?

```graphql
mutation {
  chargePaymentMethod(input: {
    paymentMethodId: "fake-valid-nonce",
    transaction: {
      amount: "2001.00"
    }
  }) {
    transaction {
      id
      createdAt
      legacyId
      status
      processorResponse {
        legacyCode
        message
      }
    }
  }
}
```

**Comment by alexgenco** [comment]
You will have to use [fragments](https://graphql.org/learn/queries/#fragments) to get the decline processor response. Building off your example:

```graphql
mutation {
  chargePaymentMethod(input: {
    paymentMethodId: "fake-valid-nonce",
    transaction: {
      amount: "2001.00"
    }
  }) {
    transaction {
      id
      status
      statusHistory {
        ... on ProcessorDeclinedEvent {
          processorResponse {
            legacyCode
            message
          }
        }
      }
    }
  }
}
```

This is because the `statusHistory` field returns `PaymentStatusEvent`, which is an interface, and not all the implementers of this interface have the `processorResponse` field (because they don't all result from backend processor interactions).

So if you want to pull the processor response for other types of events, you will need to add an additional fragment (i.e. `... on AuthorizedEvent { ... }`).

The `statusHistory` response field will be a list of objects, and the one corresponding to this `ProcessorDeclinedEvent` will contain a `processorResponse` object with the selected fields.

Here is our reference docs for the `PaymentStatusEvent` interface: https://graphql.braintreepayments.com/reference/#interface--paymentstatusevent

Hopefully that clarifies. And thanks for pointing out that typo, we will fix it in our next release.

**Comment by lkorth** [comment]
Just released 41666a2d51255f641a45d340fe6a9f5a070598f1
